storage/Devices/PartitionImpl.cc

```diff
@@ -481,13 +518,57 @@ namespace storage
     Text
     Partition::Impl::do_set_id_text(Tense tense) const
     {
+	const PartitionTable* partition_table = get_partition_table();
+
 	string tmp = id_to_string(get_id());
 
-	if (tmp.empty())
-	    return sformat(_("Set id of partition %1$s to 0x%2$02X"), get_name().c_str(), get_id());
+	if (partition_table->get_impl().is_partition_id_value_meaningful())
```

Comment:
Indeed, something like `is_partition_id_standard()` is more obvious to me. Anyway, I'm fine with the current name as long as the explanation in your former comment is anywhere in the code. Your comment is way more explanatory than the inline documentation of `is_partition_id_value_meaningful`.